infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -105,13 +94,17 @@
      * This cache services requests for data at various addresses. In a memory implementation,
      * it is not backed by anything, but in a disk implementation it is backed by persistent storage.
      */
-    private final LoadingCache<LogAddress, LogData> dataCache;
+    final LoadingCache<LogAddress, LogData> dataCache;
     private final long maxCacheSize;
 
     private final StreamLog streamLog;
 
     private final BatchWriter<LogAddress, LogData> batchWriter;
 
+    private LogUnitServerQuorumReplicationDelegate quorumDelegate = new LogUnitServerQuorumReplicationDelegate(this);
+    private LogUnitServerReplexReplicationDelegate replexDelegate = new LogUnitServerReplexReplicationDelegate(this);
+
```

Comment:
Agree. I just didn't want to make too many changes in the  review, as this changeset became epic.